geode-core/src/test/java/com/gemstone/gemfire/cache/query/QueryWithBucketParameterIntgTest.java

```diff
@@ -0,0 +1,224 @@
+        }
+        return set;
+    }
+
+    private PartitionAttributesFactory getPartitionAttributesFactoryWithPartitionResolver(int totalBuckets) {
+        PartitionAttributesFactory pAFactory = new PartitionAttributesFactory();
+        pAFactory.setRedundantCopies(1).setTotalNumBuckets(totalBuckets).setPartitionResolver(
+                getPartitionResolver());
+        return pAFactory;
+    }
+
+    private PartitionResolver getPartitionResolver() {
+        return new PartitionResolver() {
+
+            public String getName()
+            {
+                return "PartitionResolverForTest";
+            }
+
+            public Serializable getRoutingObject(EntryOperation opDetails)
+            {
+                return (Serializable)opDetails.getKey();
+            }
+
+            public void close()
+            {
+
+            }
+
+        };
+    }
+
+    private void populateRegion(Region region, int numValues) {
+        for (int i = 1; i <= numValues; i++) {
+            region.put(i, new MyValue(i));
+
+        }
+    }
+
+    /**
+     * Post test completion clean-up/tear down of the test bed
+     * This code snippet has been borrowed from (com.gemstone.gemfire.cache.query) BugJUintTest.java [03/02/2016]
+     * @throws Exception
+     */
+    @After
+    public void tearDown() throws Exception {
+        CacheUtils.closeCache();
+    }
+
+    /**
+     * remark: Query is executed with an empty bucket parameter
+     * expected behaviour : A query executed on empty hashset as its bucket set must return no result.
+     * @throws Exception
+     */
+    @Test
+    public void testQueryExecuteWithEmptyBucketListExpectNoResults() throws Exception
+    {
+
+        SelectResults r = (SelectResults)lds.executeQuery(queryExecutor, null, new HashSet<Integer>());
+        assertTrue("Received: A non-empty result collection, expected : Empty result collection", r.isEmpty());
+
+    }
+    /**
+     * remark: Query is executed with an null bucket parameter
+     * expected behaviour : the query is executed on all of its available bucket and set must return no result.
+     * null bucket parameter will set the engine to operate in default mode, which means all of regions available
+     * buckets. Hence result should not be empty.
+     * @throws Exception
+     */
+    @Test
+    public void testQueryExecuteWithNullBucketListExpectNonEmptyResultSet() throws Exception
+    {
+
+        SelectResults r = (SelectResults)lds.executeQuery(queryExecutor, null, null);
+        assertFalse("Received: An empty result collection, expected : Non-empty result collection", r.isEmpty());
+
+    }
+    /**
+     * remark: Query is executed with an valid non-empty bucket parameter
+     * expected behaviour : the query is executed on valid buckets hence returns valid non empty result.
+     * @throws Exception
+     */
+    @Test
+    public void testQueryExecuteWithNonEmptyBucketListExpectNonEmptyResultSet() throws Exception
```

Comment:
The test names are pretty self explanatory, good job!  I think with that, the comments on the test are kind of redundant.